lldb/source/Expression/IRInterpreter.cpp

```diff
@@ -1564,7 +1660,103 @@ bool IRInterpreter::Interpret(llvm::Module &module, llvm::Function &function,
+
+      uint64_t index = index_scalar.ULongLong();
+
+      // Get the vector type information
+      auto *vector_type = dyn_cast<FixedVectorType>(vector_operand->getType());
+      if (!vector_type) {
+        LLDB_LOGF(log, "ExtractElement instruction doesn't have a fixed vector "
+                       "operand type");
+        error =
+            lldb_private::Status::FromErrorString(interpreter_internal_error);
+        return false;
+      }
+
+      unsigned num_elements = vector_type->getNumElements();
+      if (index >= num_elements) {
+        LLDB_LOGF(log,
+                  "ExtractElement index %llu is out of bounds for vector with "
+                  "%u elements",
+                  (unsigned long long)index, num_elements);
+        error = lldb_private::Status::FromErrorString(bad_value_error);
+        return false;
+      }
+
+      Type *element_type = vector_type->getElementType();
+      size_t element_size = data_layout.getTypeStoreSize(element_type);
+
+      // Handle target-specific vector element ordering
+      bool reverse_elements = !MemoryMatchesIRElementOrder(exe_ctx);
+      uint64_t target_index =
+          reverse_elements ? (num_elements - 1 - index) : index;
+      size_t element_offset = target_index * element_size;
+
+      // Allocate space for the result element
+      lldb::addr_t result_addr =
+          frame.ResolveValue(extract_inst, module, exe_ctx);
+      if (result_addr == LLDB_INVALID_ADDRESS) {
+        LLDB_LOGF(log, "ExtractElement's result doesn't resolve to anything");
+        error = lldb_private::Status::FromErrorString(bad_value_error);
+        return false;
+      }
+
+      // Read the element from the vector
+      lldb_private::DataBufferHeap element_buffer(element_size, 0);
+      lldb_private::Status read_error;
+      execution_unit.ReadMemory(element_buffer.GetBytes(),
+                                vector_addr + element_offset, element_size,
+                                read_error);
+      if (!read_error.Success()) {
+        LLDB_LOGF(log, "Couldn't read element data for ExtractElement");
+        error = lldb_private::Status::FromErrorString(memory_read_error);
+        return false;
+      }
+
+      // Write the element to the result location
+      lldb_private::Status write_error;
+      execution_unit.WriteMemory(result_addr, element_buffer.GetBytes(),
+                                 element_size, write_error);
+      if (!write_error.Success()) {
+        LLDB_LOGF(log, "Couldn't write result for ExtractElement");
+        error = lldb_private::Status::FromErrorString(memory_write_error);
+        return false;
+      }
+
+      if (log) {
+        LLDB_LOGF(log, "Interpreted an ExtractElement");
+        LLDB_LOGF(log, "  Vector: 0x%" PRIx64, vector_addr);
+        LLDB_LOGF(log, "  Index: %llu", (unsigned long long)index);
+        LLDB_LOGF(log, "  Element offset: %zu", element_offset);
+        LLDB_LOGF(log, "  Result: 0x%" PRIx64, result_addr);
```

Comment:
Can you combin these into a single `LLDB_LOG`? You can replace the printf specificers with `{0}, {1}, etc.` too